src/psyclone/transformations.py

```diff
@@ -250,6 +251,33 @@ def validate(self, node, options=None):
                     "Cannot apply COLLAPSE({0}) clause to a loop nest "
                     "containing only {1} loops".format(collapse, loop_count))
 
+        # Check that there are no loop-carried dependencies
+        dep_tools = DependencyTools()
+
+        try:
+            if not dep_tools.can_loop_be_parallelised(node,
+                                                      only_nested_loops=False):
+
+                # The DependencyTools also returns False for things that are
+                # not an issue, so we ignore specific messages.
+                for message in dep_tools.get_all_messages():
+                    if "is only written once." in message:
+                        continue
+                    if "which indicates a reduction." in message:
+                        continue
```

Comment:
Shouldn't we reject this unless the user specifies that a reduction should be performed?

Reply:
(Not that we support that currently but otherwise the parallel code won't behave as expected?)

Reply:
Aha, I thought LFRic had them because one test was failing with this message but after further inspection this test had no reductions. There was a bug in the dependency tools were a READWRITE was not considered a WRITE and therefore there was false positives when figuring out if something was a reduction. This should be fixed now.